modules/swagger-codegen/src/main/resources/python/api_client.mustache

```diff
@@ -132,14 +142,15 @@ class ApiClient(object):
             post_params = self.parameters_to_tuples(post_params,
                                                     collection_formats)
 
-        # auth setting
-        self.update_params_for_auth(header_params, query_params, auth_settings)
-
+ 
         # body
         if body:
             body = self.sanitize_for_serialization(body)
 
-        # request url
+        # auth setting
+        self.update_params_for_auth(header_params, query_params, auth_settings, resource_path, body)
```

Comment:
I don't see why this line should be moved here.

Reply:
Have to compute signature after all the data is there.